src/libraries/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -240,7 +240,7 @@ public Icon(string fileName)
         public Icon(Type type, string resource)
         {
             if (resource == null)
-                throw new ArgumentException(nameof(resource));
+                throw new ArgumentNullException(nameof(resource));
```

Comment:
` AssertExtensions.Throws<ArgumentException>(null, () => new Icon(typeof(Icon), null));` test is failing as the Windows implementation of this method is not checking this condition and not throwing, instead directly calling `type.GetTypeInfo().Assembly.GetManifestResourceStream(type, resource)` which is called for Unix implementation after these checks in row 249.

This kind of pattern happening for most of the `Unix` implementation updates in this PR and causing test failures. I think `parameterName` update for ArgumentExceptions is fine, i can fix tests by checking the OS, but is it OK to throw different exceptions (ArgumentNullException instead ArgumentException) for the same method but different OS? 
CC @bartonjs @stephentoub

Reply:
Absent other data, it sounds like we should change the Windows implementation to throw the appropriate ArgumentNullException.